ads/google/a4a/utils.js

```diff
@@ -113,9 +114,27 @@ export function googleAdUrl(
     const viewportRect = viewport.getRect();
     const iframeDepth = iframeNestingDepth(win);
     const viewportSize = viewport.getSize();
-    if (ValidAdContainerTypes.indexOf(adElement.parentElement.tagName) >= 0) {
-      queryParams.push({name: 'amp_ct',
-                        value: adElement.parentElement.tagName});
+
+    // Detect container types.
+    let parentElement = adElement.parentElement;
+    let tagName = parentElement.tagName.toUpperCase();
+    const containerTypeSet = {};
+    while (parentElement && ValidAdContainerTypes[tagName]) {
```

Comment:
To be clear, as soon as you hit an element that is not one of the container types, you will exit this loop.  For instance:

<amp-sticky-ad>
   <div>
     <amp-lightbox>
        <amp-ad>

You would not find amp-sticky-ad.  Is this intentional?